packages/design-system/src/components/primitives/arrow-focus.tsx

```diff
@@ -0,0 +1,162 @@
+import type { KeyboardEvent } from "react";
+
+const focusableSelector = (extra: string) =>
+  [
+    "input:not([type=hidden])",
+    "select",
+    "textarea",
+    "button",
+    "a[href]",
+    "area[href]",
+    "summary",
+    "iframe",
+    "object",
+    "embed",
+    "audio[controls]",
+    "video[controls]",
+    "[contenteditable]",
+    "[tabindex]",
+  ]
+    .map((selector) => `${selector}:not([disabled]):not([hidden])${extra}`)
```

Comment:
Never seen `hidden` attribute in a wild.

Also instead of `:not[disabled])` you can just use  
```
  'button:enabled',
  'select:enabled',
  'textarea:enabled',
  'input:enabled',
```
in your list

Reply:
hidden is basically html version of display: none. Allows to not rebuild dom.

Reply:
Sick, never knew there is :enabled